rllib/core/optim/tests/test_rl_optimizer_tf.py

```diff
@@ -0,0 +1,477 @@
+from dataclasses import dataclass, field
+import gym
+import numpy as np
+import tensorflow as tf
+import tensorflow_probability as tfp
+from typing import Any, List, Mapping, Union
+import unittest
+
+from ray.rllib.algorithms import AlgorithmConfig
+from ray.rllib.offline import IOContext
+from ray.rllib.offline.dataset_reader import (
+    DatasetReader,
+    get_dataset_and_shards,
+)
+
+from ray.rllib.core.optim.rl_optimizer import RLOptimizer
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.core.rl_module.tf.tf_rl_module import TFRLModule
+from ray.rllib.models.specs.specs_dict import ModelSpec, check_specs
+from ray.rllib.models.specs.specs_tf import TFTensorSpecs
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.test_utils import check
+from ray.rllib.utils.typing import TensorType
+
+
+def do_rollouts(
+    env: gym.Env, module_for_inference: RLModule, num_rollouts: int
+) -> SampleBatch:
+    _returns = []
+    _rollouts = []
+    for _ in range(num_rollouts):
+        obs = env.reset()
+        _obs, _next_obs, _actions, _rewards, _dones = [], [], [], [], []
+        _return = -0
+        for _ in range(env._max_episode_steps):
+            _obs.append(obs)
+            fwd_out = module_for_inference.forward_inference(
+                {"obs": tf.convert_to_tensor([obs], dtype=tf.float32)}
+            )
+            action = convert_to_numpy(fwd_out["action_dist"].sample())[0]
+            next_obs, reward, done, _ = env.step(action)
+            _next_obs.append(next_obs)
+            _actions.append([action])
+            _rewards.append([reward])
+            _dones.append([done])
+            _return += reward
+            if done:
+                break
+            obs = next_obs
+        batch = SampleBatch(
+            {
+                "obs": _obs,
+                "next_obs": _next_obs,
+                "actions": _actions,
+                "rewards": _rewards,
+                "dones": _dones,
+            }
+        )
+        _returns.append(_return)
+        _rollouts.append(batch)
+    return np.mean(_returns), _returns, _rollouts
+
+
+@dataclass
+class FCConfig:
```

Comment:
kourosh already has these defined somewhere else right?

Reply:
yes. Let's not even have this. The BCRLModule can be even simpler than this. checkout this PR: https://github.com/ray-project/ray/pull/31069